lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -0,0 +1,336 @@
+
+test.before(async (t) => {
+  process.env.stackName = cryptoRandomString({ length: 10 });
+  process.env.system_bucket = cryptoRandomString({ length: 10 });
+
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.ProvidersTable = cryptoRandomString({ length: 10 });
+  process.env.PdrsTable = cryptoRandomString({ length: 10 });
+
+  await createBucket(process.env.system_bucket);
+
+  collectionsModel = new Collection();
+  await collectionsModel.createTable();
+
+  providersModel = new Provider();
+  await providersModel.createTable();
+
+  pdrsModel = new Pdr();
+  await pdrsModel.createTable();
+
+  t.context.knexAdmin = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      migrationDir,
+    },
+  });
+  await t.context.knexAdmin.raw(`create database "${testDbName}";`);
+  await t.context.knexAdmin.raw(`grant all privileges on database "${testDbName}" to "${testDbUser}"`);
+
+  t.context.knex = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      PG_DATABASE: testDbName,
+      migrationDir,
+    },
+  });
+
+  await t.context.knex.migrate.latest();
+});
+
+test.beforeEach(async (t) => {
+  const testCollection = fakeCollectionFactory();
+  const testProvider = fakeProviderFactory({
+    encrypted: true,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+  });
+
+  t.context.testCollection = testCollection;
+  t.context.testProvider = testProvider;
+
+  t.context.collectionCumulusId = await migrateFakeCollectionRecord(testCollection, t.context.knex);
+  t.context.providerCumulusId = await migrateFakeProviderRecord(testProvider, t.context.knex);
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex(tableNames.pdrs).del();
+  await t.context.knex(tableNames.providers).del();
+  await t.context.knex(tableNames.collections).del();
+});
+
+test.after.always(async (t) => {
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+
+  await providersModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await pdrsModel.deleteTable();
+
+  await t.context.knex.destroy();
+  await t.context.knexAdmin.raw(`drop database if exists "${testDbName}"`);
+  await t.context.knexAdmin.destroy();
+});
+
+test.serial('migratePdrRecord correctly migrates PDR record', async (t) => {
+  const { knex, testCollection, testProvider, collectionCumulusId, providerCumulusId } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+  await migratePdrRecord(testPdr, knex);
+
+  const record = await t.context.knex.queryBuilder()
+    .select()
+    .table(tableNames.pdrs)
+    .where({ name: testPdr.pdrName })
+    .first();
+
+  t.like(
+    omit(record, ['cumulus_id']),
+    omit({
+      name: testPdr.pdrName,
+      provider_cumulus_id: providerCumulusId,
+      collection_cumulus_id: collectionCumulusId,
+      execution_cumulus_id: testPdr.executionCumulusId ? testPdr.executionCumulusId : null,
```

Comment:
It seems like we know that `testPdr.executionCumulusId` is undefined always for this test, so we can always expect this to be `null`? I don't think we need the ternary expression

Reply:
Yeah, at that point in my code executionCumulusId was always undefined but I've seen added in at least one example where it's not undefined. Thanks for bringing this to my attention! It reminded me that I wanted to add at least one example where it was defined.